lib/Demangling/Demangler.cpp

```diff
@@ -1875,6 +1875,20 @@ NodePointer Demangler::popAssocTypePath() {
   return AssocTypePath;
 }
 
+NodePointer Demangler::popSelfOrAssocTypePath() {
+  if (auto Type = popNode(Node::Kind::Type)) {
+    if (Type->getChild(0) &&
+        Type->getChild(0)->getKind() ==
+        Node::Kind::DependentGenericParamType) {
+      return Type;
+    }
+
+    pushNode(Type);
+  }
+
+  return popAssocTypePath();
```

Comment:
Yeah, I mean, I don't know if reducing the number of mangling operators really justifies the complexity and risk of future conflict here.

Reply:
ok